test/thermodynamics/efficiency_factor_test.py

```diff
@@ -0,0 +1,34 @@
+from collections import namedtuple
+from pytest import approx, fixture, raises
+from symplyphysics import (errors, units, convert_to, Quantity, SI, dimensionless)
+from symplyphysics.laws.thermodynamics import efficiency_factor as efficiency_law
+
+# Description
+## Suppose we have a heat engine that receives 2 Joules of heat from the heater, and gives 1.5 Joules to the refrigerator. Then its efficiency should be (2 - 1.5) / 2 = 0.25
+
+
+@fixture(name="test_args")
+def test_args_fixture():
+    Q_h = Quantity(2 * units.joule)
+    Q_r = Quantity(1.5 * units.joule)
+    Args = namedtuple("Args", ["Q_h", "Q_r"])
+    return Args(Q_h=Q_h, Q_r=Q_r)
+
+
+def test_basic_efficiency(test_args):
+    result = efficiency_law.calculate_efficiency_factor(test_args.Q_h, test_args.Q_r)
+    assert SI.get_dimension_system().equivalent_dims(result.dimension, dimensionless)
```

Comment:
Now when you have `float` as a result, you do not need to check resulting dimension, and you do not need to convert.